cmd/replicate-reset-status.go

```diff
@@ -0,0 +1,161 @@
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	humanize "github.com/dustin/go-humanize"
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	json "github.com/minio/colorjson"
+	"github.com/minio/mc/pkg/probe"
+	"github.com/minio/minio-go/v7/pkg/replication"
+	"github.com/minio/pkg/console"
+)
+
+var replicateResyncStatusFlags = []cli.Flag{
+	cli.StringFlag{
+		Name:  "remote-bucket",
+		Usage: "remote bucket ARN",
+	},
+}
+
+var replicateResyncStatusCmd = cli.Command{
+	Name:         "status",
+	Usage:        "status of replication resync",
+	Action:       mainreplicateResyncStatus,
+	OnUsageError: onUsageError,
+	Before:       setGlobalsFromContext,
+	Flags:        append(globalFlags, replicateResyncStatusFlags...),
+	CustomHelpTemplate: `NAME:
+   {{.HelpName}} - {{.Usage}}
+
+USAGE:
+   {{.HelpName}} TARGET
+
+FLAGS:
+   {{range .VisibleFlags}}{{.}}
+   {{end}}
+EXAMPLES:
+  1. Status of replication resync in bucket "mybucket" for alias "myminio" for all targets.
+   {{.Prompt}} {{.HelpName}} myminio/mybucket
+
+  2. Status of replication resync in bucket "mybucket" for remote bucket target.
+   {{.Prompt}} {{.HelpName}} myminio/mybucket --remote-bucket "arn:minio:replication::xxx:mybucket"
+`,
+}
+
+// checkreplicateResyncStatusSyntax - validate all the passed arguments
+func checkreplicateResyncStatusSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) != 1 {
+		cli.ShowCommandHelpAndExit(ctx, "status", 1) // last argument is exit code
+	}
+}
+
+type replicateResyncStatusMessage struct {
+	Op                string                        `json:"op"`
+	URL               string                        `json:"url"`
+	ResyncTargetsInfo replication.ResyncTargetsInfo `json:"resyncInfo"`
+	Status            string                        `json:"status"`
+	TargetArn         string                        `json:"targetArn"`
+}
+
+func (r replicateResyncStatusMessage) JSON() string {
+	r.Status = "success"
+	jsonMessageBytes, e := json.MarshalIndent(r, "", " ")
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+	return string(jsonMessageBytes)
+}
+
+func (r replicateResyncStatusMessage) String() string {
+	if len(r.ResyncTargetsInfo.Targets) == 0 {
+		return console.Colorize("replicateResyncStatusWarn", "No replication resync status available.")
+	}
+	coloredDot := console.Colorize("Headers", dot)
+	var rows string
+	rows += console.Colorize("TDetail", "Resync status summary:")
+
+	for _, st := range r.ResyncTargetsInfo.Targets {
+		rows += "\n"
+		rows += console.Colorize("replicateResyncStatusMsg", newPrettyTable(" | ",
+			Field{"ARN", 120},
+		).buildRow(fmt.Sprintf("%s %s", coloredDot, st.Arn)))
+		rows += "\n"
+		rows += console.Colorize("TDetail", "   Status: ")
+		rows += console.Colorize(st.ResyncStatus, st.ResyncStatus)
+		rows += "\n"
+
+		var maxLen = 15
```

Comment:
To keep the formatting for all sized terminals you should use `console.NewTable()`